src/database/methods.rs

```diff
@@ -239,70 +252,57 @@ pub trait DbMethods<'c>: Acquire<'c, Database = Postgres> + Sized {
         .await?)
     }
 
-    #[instrument(skip(self), level = "debug")]
-    async fn get_commitments_by_statuses(
-        self,
-        statuses: Vec<ProcessedStatus>,
-    ) -> Result<Vec<TreeUpdate>, Error> {
+    async fn get_tree_updates_after_id(self, id: usize) -> Result<Vec<TreeUpdate>, Error> {
         let mut conn = self.acquire().await?;
 
-        let statuses: Vec<&str> = statuses.into_iter().map(<&str>::from).collect();
         Ok(sqlx::query_as::<_, TreeUpdate>(
             r#"
             SELECT id as sequence_id, leaf_index, commitment as element, root as post_root
             FROM identities
-            WHERE status = ANY($1)
+            WHERE id > $1
             ORDER BY id ASC;
             "#,
         )
-        .bind(&statuses[..]) // Official workaround https://github.com/launchbadge/sqlx/blob/main/FAQ.md#how-can-i-do-a-select--where-foo-in--query
+        .bind(id as i64)
         .fetch_all(&mut *conn)
         .await?)
     }
 
-    async fn get_commitments_after_id(self, id: usize) -> Result<Vec<TreeUpdate>, Error> {
+    async fn get_latest_tree_update_by_statuses(
+        self,
+        statuses: Vec<ProcessedStatus>,
+    ) -> Result<Option<TreeUpdate>, Error> {
         let mut conn = self.acquire().await?;
 
+        let statuses: Vec<&str> = statuses.into_iter().map(<&str>::from).collect();
         Ok(sqlx::query_as::<_, TreeUpdate>(
             r#"
             SELECT id as sequence_id, leaf_index, commitment as element, root as post_root
             FROM identities
-            WHERE id > $1
-            ORDER BY id ASC;
+            WHERE status = ANY($1)
+            ORDER BY id DESC
+            LIMIT 1;
             "#,
         )
-        .bind(id as i64)
-        .fetch_all(&mut *conn)
+        .bind(&statuses[..]) // Official workaround https://github.com/launchbadge/sqlx/blob/main/FAQ.md#how-can-i-do-a-select--where-foo-in--query
+        .fetch_optional(&mut *conn)
         .await?)
     }
 
-    #[instrument(skip(self, leaf_indexes), level = "debug")]
-    async fn get_non_zero_commitments_by_leaf_indexes<I: IntoIterator<Item = usize>>(
-        self,
-        leaf_indexes: I,
-    ) -> Result<Vec<Hash>, Error>
-    where
-        I: IntoIterator<Item = usize> + Send,
-    {
+    async fn get_tree_update_by_root(self, root: &Hash) -> Result<Option<TreeUpdate>, Error> {
         let mut conn = self.acquire().await?;
 
-        let leaf_indexes: Vec<i64> = leaf_indexes.into_iter().map(|v| v as i64).collect();
-
-        Ok(sqlx::query(
+        Ok(sqlx::query_as::<_, TreeUpdate>(
```

Comment:
Redundant type specification? This should without the `::<_, TreeUpdate>`

```suggestion
        Ok(sqlx::query_as(
```